rust-lang/cargo
# Report cmd aliasing failure with more contexts

### What does this PR try to resolve?

Fixes #11013.

### How should we test and review this PR?

Add an extra `ConfigErrorKind` type on `ConfigError` to enable caller to check which kind of error it is.

### Additional information

It's better but still not optimal. For example, with this config

```toml
[alias]
foo = "clippy"
clippy = [1]
```

and run `cargo foo`, it didn't attach `clippy` on the error message of aliases chain. That's because TOML parsing error occurred before expanding `clippy` alias successfully.
